### PR TITLE
Phase 3 Task 15 L2 Hello World: justify-content main-axis alignment

### PR DESCRIPTION
**Pull request by raroche**
## Summary

- **CSS Box Alignment L3 §4.5.** FlexLayouter now honors `justify-content` main-axis alignment — adds support for `flex-end`, `center`, `space-between`, `space-around`, `space-evenly` (L1 already supported the implicit `flex-start` default).
- **Algorithm**: pre-pass computes `freeSpace = contentInlineSize - totalItemInlineSize`; resolves keyword to `(startOffset, betweenSpacing)`; cursor loop becomes `cursor = contentInlineOffset + startOffset; cursor += itemInlineSize + betweenSpacing`.
- **Spec edge cases honored**: overflow → flex-start fallback; N=1 with `space-between` → flex-start fallback; logical aliases (start/end/left/right) → flex-start/flex-end for `flex-direction: row` (L1 default).
- **+9 tests** (8 unit + 1 production-pipeline). Zero existing-test regressions.

## Test plan

- [x] `dotnet build NetPdf.slnx -c Release` — 0 warnings, 0 errors
- [x] `dotnet test tests/NetPdf.UnitTests/NetPdf.UnitTests.csproj -c Release --nologo` — 4417 passed (+9 over PR #61 baseline of 4408), 4 skipped, 0 failed
- [x] `./scripts/aot-parity.sh` — 4/4 parity tests passed
- [x] `git diff --check` — clean

## Deferred to L3+

- `safe` / `unsafe` overflow-position modifiers for `justify-content`
- Writing-mode-aware `left` / `right` mapping (L2 maps to flex-start/flex-end for row direction only)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/NetPdf.Layout/Layouters/FlexLayouter.cs [hunk 1]
```diff
@@ -329,6 +377,61 @@ public LayoutAttemptResult AttemptLayout(
...
+    private static (double startOffset, double betweenSpacing) ComputeJustifyContentOffsets(
+        JustifyContentValue justifyContent, double freeSpace, int itemCount)
+    {
+        if (itemCount == 0 || freeSpace <= 0)
+        {
+            // No items OR overflow — flex-start fallback (cursor stays
+            // at the container's content-inline-start, no extra
+            // between-item spacing).
+            return (0, 0);
+        }
```
[the remark this answers is not in the document]
Comment (raroche) [reply src/NetPdf.Layout/Layouters/FlexLayouter.cs]: Resolved by F#2 in hardening commit 59722e0. `ComputeJustifyContentOffsets` now branches by value family per CSS Box Alignment L3 §5.3: positional values (flex-start/flex-end/center) keep their natural offset on overflow (allowing items to overflow equally for center, etc.); distribution values fall back to safe-start. `safe X` modifier always falls back to safe-start; `unsafe X` modifier always honors the natural offset.

File: src/NetPdf.Layout/Layouters/FlexLayouter.cs [hunk 2]
```diff
@@ -329,6 +377,61 @@ public LayoutAttemptResult AttemptLayout(
...
+            JustifyContentValue.Center => (freeSpace / 2.0, 0),
+            JustifyContentValue.SpaceBetween => itemCount >= 2
+                ? (0, freeSpace / (itemCount - 1))
+                : (0, 0), // N=1 → flex-start fallback per spec
+            JustifyContentValue.SpaceAround => (
+                freeSpace / (2.0 * itemCount),
+                freeSpace / itemCount),
+            JustifyContentValue.SpaceEvenly => (
+                freeSpace / (itemCount + 1),
+                freeSpace / (itemCount + 1)),
```
[the remark this answers is not in the document]
Comment (raroche) [reply src/NetPdf.Layout/Layouters/FlexLayouter.cs]: Resolved by F#2 in hardening commit 59722e0. `ComputeJustifyContentOffsets` now branches by value family per CSS Box Alignment L3 §5.3: positional values (flex-start/flex-end/center) keep their natural offset on overflow (allowing items to overflow equally for center, etc.); distribution values fall back to safe-start. `safe X` modifier always falls back to safe-start; `unsafe X` modifier always honors the natural offset.

File: tests/NetPdf.UnitTests/Phase3/FlexLayouterTests.cs [hunk 3]
```diff
@@ -233,6 +233,233 @@ public void Flex_layouter_rejects_non_flex_root_box()
...
+        // 0, 100, 200, 300, 400.
+        var fragments = LayoutNItemsWithJustifyContent(
+            keywordIndex: 5, itemWidth: 100, containerWidth: 300, itemCount: 5);
+
+        Assert.Equal(5, fragments.Count);
+        Assert.Equal(0.0, fragments[0].InlineOffset, precision: 3);
+        Assert.Equal(100.0, fragments[1].InlineOffset, precision: 3);
+        Assert.Equal(200.0, fragments[2].InlineOffset, precision: 3);
+        Assert.Equal(300.0, fragments[3].InlineOffset, precision: 3);
+        Assert.Equal(400.0, fragments[4].InlineOffset, precision: 3);
```
[the remark this answers is not in the document]
Comment (raroche) [reply tests/NetPdf.UnitTests/Phase3/FlexLayouterTests.cs]: Resolved by F#2 in hardening commit 59722e0. The old `L2_justify_content_overflow_falls_back_to_flex_start` test (which pinned the buggy behavior) was rewritten in-place to `L2_justify_content_center_with_overflow_keeps_centered_negative_offset` — asserts the spec-correct behavior where center with negative free-space places items at `freeSpace/2` (= negative), so items overflow equally on both sides. Added complementary `L2_justify_content_space_between_with_overflow_falls_back_to_flex_start` test that pins the distribution-family fallback.

[the remark this answers is not in the document]
Comment (raroche) [reply tests/NetPdf.UnitTests/Phase3/FlexLayouterTests.cs]: Resolved by F#2 in hardening commit 59722e0. The old `L2_justify_content_overflow_falls_back_to_flex_start` test (which pinned the buggy behavior) was rewritten in-place to `L2_justify_content_center_with_overflow_keeps_centered_negative_offset` — asserts the spec-correct behavior where center with negative free-space places items at `freeSpace/2` (= negative), so items overflow equally on both sides. Added complementary `L2_justify_content_space_between_with_overflow_falls_back_to_flex_start` test that pins the distribution-family fallback.

File: src/NetPdf.Layout/Layouters/ComputedStyleLayoutExtensions.cs [hunk 4]
```diff
@@ -403,6 +403,65 @@ public static ColumnFillValue ReadColumnFill(this ComputedStyle style)
...
+            3 => JustifyContentValue.SpaceEvenly,
+            4 => JustifyContentValue.FlexStart,       // stretch → flex-start for flex per spec
+            // <content-position>
+            5 => JustifyContentValue.Center,
+            6 => JustifyContentValue.FlexStart,       // start → flex-start (LTR row)
+            7 => JustifyContentValue.FlexEnd,         // end → flex-end (LTR row)
+            8 => JustifyContentValue.FlexStart,
+            9 => JustifyContentValue.FlexEnd,
+            10 => JustifyContentValue.FlexStart,      // left → flex-start (row)
+            11 => JustifyContentValue.FlexEnd,        // right → flex-end (row)
```
[the remark this answers is not in the document]
Comment (raroche) [reply src/NetPdf.Layout/Layouters/ComputedStyleLayoutExtensions.cs]: Resolved by follow-up commit c11cafa. Added 2 new `[Theory]` tests covering all 12 alias data-rows: `L2_justify_content_logical_alias_maps_to_physical_for_ltr_row` (4 rows: start/end/left/right → flex-start/flex-end for LTR row) + `L2_justify_content_safe_unsafe_alias_compounds_map_to_physical_without_overflow` (8 rows: safe X / unsafe X compounds for the same 4 aliases without overflow). Pins the L1-default-row-only contract; writing-mode-aware mapping deferred to L3+.

File: src/NetPdf.Layout/Layouters/ComputedStyleLayoutExtensions.cs [hunk 5]
```diff
@@ -403,6 +403,65 @@ public static ColumnFillValue ReadColumnFill(this ComputedStyle style)
...
+            // safe / unsafe <position> compounds (indices 12+) — L2
+            // treats them as flex-start; the overflow modifier's safe-
+            // mode containment is L3+ scope. The bare-position semantics
+            // are captured when authors omit the overflow modifier.
+            _ => JustifyContentValue.FlexStart,       // 0 (normal) → flex-start; unknown → flex-start
```
[the remark this answers is not in the document]
Comment (raroche) [reply src/NetPdf.Layout/Layouters/ComputedStyleLayoutExtensions.cs]: Resolved by follow-up commit c11cafa. Added 2 new `[Theory]` tests covering all 12 alias data-rows: `L2_justify_content_logical_alias_maps_to_physical_for_ltr_row` (4 rows: start/end/left/right → flex-start/flex-end for LTR row) + `L2_justify_content_safe_unsafe_alias_compounds_map_to_physical_without_overflow` (8 rows: safe X / unsafe X compounds for the same 4 aliases without overflow). Pins the L1-default-row-only contract; writing-mode-aware mapping deferred to L3+.

File: PROGRESS.md [hunk 6]
```diff
@@ -1,13 +1,15 @@
...
...nt: space-between; width: 600px">` with three 50px-wide `<span>` items → offsets 0/275/550 through the full HTML → cascade → BoxBuilder → BlockLayouter → FlexLayouter pipeline. **Diff:** 5 files changed, ~518 insertions, ~17 deletions. **`docs/deferrals.md` updated**: `flex-layouter-features` Missing list now reflects L2's progress — removed the bare "`justify-content` values beyond `flex-start`" bullet; added the still-deferred edge cases (`safe`/`unsafe` overflow-position modifiers; writing-mode-aware `left`/`right` mapping). **Flex L3+ scope (updated):** `flex-direction: column` / `row-reverse` / `column-reverse`; `flex-wrap: wrap` / `wrap-reverse`; `align-items` values beyond default stretch; `flex-grow` / `flex-shrink` / `flex-basis`; `order` property; baseline alignment; multi-page flex container splitting (atomic to outer pagination in L1+L2); inline-flex-item anonymous wrapping for stray TextRun whitespace; `safe`/`unsafe` overflow-position modifiers for `justify-content`; writing-mode-aware logical-axis aliases. Active-task pointer advances to **Task 15 L3** (`align-items` cross-axis OR `flex-direction: column` OR `flex-wrap: wrap`) OR revisit multicol cycle 5+ deferrals.)
```
[the remark this answers is not in the document]
Comment (raroche) [reply PROGRESS.md]: Resolved by follow-up commit c11cafa — fixed the PROGRESS.md wording to say three <div> block children (matching the actual test at `FlexLayouterProductionTests.cs:L2_production_html_justify_content_space_between`).

[the remark this answers is not in the document]
Comment (raroche) [reply PROGRESS.md]: Resolved by F#1 in hardening commit 59722e0. The new PROGRESS.md entry correctly says "the 14 `safe X` / `unsafe X` compounds" (= 7 ContentPositions × 2 overflow modes = 14).

File: docs/deferrals.md [hunk 7]
```diff
@@ -853,7 +853,14 @@ grepping the ID).
...
   - `flex-wrap: wrap` / `wrap-reverse`
-  - `justify-content` values beyond `flex-start`
+  - `safe` / `unsafe` overflow-position modifiers for
+    `justify-content` (L2 treats compound keywords like
+    `safe center` as the bare position; the spec's safe-mode
+    overflow containment is L3+ scope)
+  - Writing-mode-aware `left` / `right` mapping for
+    `justify-content` (L2 maps both to `flex-start` / `flex-end`
+    under the L1 default LTR + `flex-direction: row`; L3+ will
+    resolve against the box's writing-mode + direction)
```
[the remark this answers is not in the document]
Comment (raroche) [reply docs/deferrals.md]: Resolved by F#5 in hardening commit 59722e0. The `flex-layouter-features` Behavior section was rewritten to describe L2's full justify-content matrix (6 base values × 3 overflow modes). Stale "items packing from start edge only" wording removed. Added the F#3 outer-main-size deferral bullet as a new Missing item.
